### ISSUE TITLE
[IOS] Chrome/Safari fails with Invalid UTF-8 sequence in header value

### ISSUE THREAD

**Issue by jfarcand** [Atmosphere/atmosphere-javascript#136]
Seems to be a bug in Chrome/Safari:
```js
WebSocket connection to 'ws://10.0.1.12:8080/chat?X-Atmosphere-tracking-id=af07a8aa-64e3-4b6d-b9c4-c1c0e3e37824&X-Atmosphere-Framework=2.2.6-javascript&X-Atmosphere-Transport=websocket&X-Atmosphere-TrackMessageSize=true&Content-Type=application/json&X-atmo-protocol=true' failed: Invalid UTF-8 sequence in header value
```
This happens when the offline/online mode is triggered. This is a side effect of  #134

**Comment by ookke** [comment]
@jfarcand:
We encountered this issue also and it seems that it's caused by a combination of https://code.google.com/p/chromium/issues/detail?id=380075 (fixed already in Blink but not in Webkit I assume) and the fact that the atmosphere backend sends an empty "X-Atmosphere-first-request" header on the first response. I added a comment to the webkit ticket also as a hint.

For now we added a workaround filter that replaces an empty first-response header with "false", but based on some quick grepping in atmosphere-javascript it seems that it doesn't even use response header directly?

Maybe atmosphere could use true/false as values for that header to work around this issue for now? Who knows how long it will take to get the webkit fix pushed to all iOS clients...

**Comment by jfarcand** [comment]
@ookke which version are you using? This has been fixed in 2.3.0 for sure. Can you try it?

**Comment by slovdahl** [comment]
FYI, it's also fixed in 2.2.5.

https://github.com/Atmosphere/atmosphere/commit/b0d9a915edba8444e5430bee1763d01843c03db9
https://github.com/Atmosphere/atmosphere/commit/e44a69aabf8e67abbd5b975a019e5c1e45b5eaca

**Comment by ookke** [comment]
@slovdahl thanks for the tip!

@jfarcand we are using 2.2.4 currently, will try upgrading to 2.2.5 first.
